packages/microsoft-entra-adapter/src/filters/app_role.ts

```diff
@@ -215,22 +229,41 @@ const deployAppRoleChangesViaParent = async ({
 
   const existingParentChangeIndices = parentChanges
     .map(({ existingParentChangeIdx }) => existingParentChangeIdx)
-    .filter(index => index !== -1)
+    .filter(isDefined)
   removeArrayElementsByIndices(otherChanges, existingParentChangeIndices)
-  const changesToDeploy = parentChanges.map(({ adjustedParentChange }) => adjustedParentChange)
-  const existingParentChanges = parentChanges.map(({ existingParentChange }) => existingParentChange).filter(isDefined)
+  const changesToDeploy = parentChanges.map(({ parentChange }) => parentChange)
 
   const calculateResult = async ({
     errors,
     appliedChanges: appliedParentChanges,
   }: DeployResult): Promise<DeployResultWithLeftoverChanges> => {
     const appliedParentChangesFullNames = appliedParentChanges.map(change => getChangeData(change).elemID.getFullName())
-    const appliedExistingParentChanges = existingParentChanges.filter(change =>
-      appliedParentChangesFullNames.includes(getChangeData(change).elemID.getFullName()),
-    )
-    const appliedAppRoleChanges = appRoleInstanceChanges.filter(change =>
-      appliedParentChangesFullNames.includes(getParents(getChangeData(change))[0]?.elemID.getFullName()),
-    )
+    const appliedExistingParentChanges = await awu(parentChanges)
+      .filter(
+        ({ parentChange, existingParentChangeIdx }) =>
+          existingParentChangeIdx !== undefined &&
+          appliedParentChangesFullNames.includes(getChangeData(parentChange).elemID.getFullName()),
+      )
+      .map(({ parentChange }) =>
+        // We add the app roles field to the parent change in place, but we don't want to copy it to the workspace.
```

Comment:
mostly a nit - 
I think it's a bit odd to mention "workspace" in here (cause it's something that the adapter is not aware of)
maybe something like  
```suggestion
        // We add the app roles field to the parent change in place, but we don't include it in the applied change.
```